src/pretix/presale/checkoutflow.py

```diff
@@ -1144,13 +1144,11 @@ def is_form_filled(form, ignore_keys=()):
 
     def _get_is_business_heuristic(self):
         key = 'checkout_heuristic_is_business:' + str(self.event.pk)
-        cached_result = caches['default'].get(key)
-        if cached_result is None:
-            if caches['default'].add(key, False, timeout=10):  # return False while query is running
+        cached_result = caches['default'].get(key, default=False)
+        if not caches['default'].get(key + ':valid'):
+            if caches['default'].add(key + ':valid', True, timeout=10):  # set valid while query is running
```

Comment:
Since `add` does nothing if the key already exists, shouldn't 

```suggestion
        if caches['default'].add(key + ':valid', True, timeout=10):  # set valid while query is running
```

also work?

Reply:
Ja stimmt, das reicht jetzt auch (als es noch nur ein key war, brauchte ich ja den inhalt davon, aber jetzt nicht mehr)